### PR TITLE
[ru] Update advanced/metaprogramming.md to 1.0.2

### PR DESCRIPTION
**Pull request by filalex77**
- Newline all the sentences to correspond to English source

### CODE REVIEW

File: ru/lessons/advanced/metaprogramming.md [hunk 1]
```diff
@@ -1,19 +1,25 @@
 ---
-version: 0.9.1
+version: 1.0.2
 title: Метапрограммирование
 ---
 
-Метапрограммирование - процесс написания кода с помощью другого кода. В Elixir это дает нам возможность расширять возможности языка для большего соответствия нуждам приложения. Мы начнем этот урок с рассмотрения внутреннего представления Elixir кода, а потом узнаем, как его модифицировать. И, в итоге, мы применим эти знания для создания собственных расширений языка.
+Метапрограммирование - процесс написания кода с помощью другого кода.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]:
```suggestion
Метапрограммирование — процесс написания кода с помощью другого кода.
```

File: ru/lessons/advanced/metaprogramming.md [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
-Предупреждение: метапрограммирование это сложно и оно должно использоваться только, когда абсолютно необходимо. Злоупотребление этим инструментом приведет к нечитабельности и сложностям в отладке кода.
+Предупреждение: метапрограммирование это сложно и оно должно использоваться только, когда абсолютно необходимо.
+Злоупотребление этим инструментом приведет к нечитабельности и сложностям в отладке кода.
 
 {% include toc.html %}
 
 ## Quote
 
-Первый шаг к пониманию метапрограммирования - понимание, что представляют собой выражения. В Elixir внутреннее представление абстрактного синтаксического дерева (AST) состоит из кортежей. Эти кортежи состоят из трех частей: название функции, метаданные и аргументы функции.
+Первый шаг к пониманию метапрограммирования - понимание, что представляют собой выражения.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]:
```suggestion
Первый шаг к пониманию метапрограммирования — понимание, что представляют собой выражения.
```

File: ru/lessons/advanced/metaprogramming.md [hunk 3]
```diff
@@ -57,15 +65,21 @@ iex> quote do: divide(42, unquote(denominator))
...
 
-В первом примере переменная `denominator` экранирована, потому результирующий AST включает кортеж для доступа к переменной. Во втором примере в результирующем коде будет только её значение.
+В первом примере переменная `denominator` экранирована, потому результирующий AST включает кортеж для доступа к переменной.
+Во втором примере в результирующем коде будет только её значение.
 
 ## Макросы
 
-Как только мы разобрались, как работают `quote/2` и `unquote/1`, мы готовы к погружению в макросы. Важно помнить, что макросы, как и все в метапрограммировании, должны использоваться очень осторожно.
+Как только мы разобрались, как работают `quote/2` и `unquote/1`, мы готовы к погружению в макросы.
+Важно помнить, что макросы, как и все в метапрограммировании, должны использоваться очень осторожно.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]:
```suggestion
Важно помнить, что макросы, как и всё в метапрограммировании, должны использоваться очень осторожно.
```

File: ru/lessons/advanced/metaprogramming.md [hunk 4]
```diff
@@ -57,15 +65,21 @@ iex> quote do: divide(42, unquote(denominator))
...
 
-В простейших терминах макросы - это специальные функции, созданные для возврата экранированных выражений, которые затем будут вставлены в код приложения. Представьте себе, что макрос заменяется значением, которое он вернет, а не будет вызван как функция. С макросами у нас есть все необходимое для расширения языка и динамического добавления кода в наше приложение.
+В простейших терминах макросы - это специальные функции, созданные для возврата экранированных выражений, которые затем будут вставлены в код приложения.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]:
```suggestion
В простейших терминах макросы — это специальные функции, созданные для возврата экранированных выражений, которые затем будут вставлены в код приложения.
```

File: ru/lessons/advanced/metaprogramming.md [hunk 5]
```diff
@@ -127,13 +144,69 @@ def test do
 end
 ```
 
+### Отладка
+
+Окей, теперь мы знаем, как использовать `quote/2`, `unquote/1` и писать макросы.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]: Maybe it makes sense to replace `Окей` with something like `Хорошо`, `Отлично` or another synonym.

File: ru/lessons/advanced/metaprogramming.md [hunk 6]
```diff
@@ -127,13 +144,69 @@ def test do
...
+Но что если у вас есть огромный кусок AST и вы хотите в нём разобраться? В таком случае вы можете использовать `Macro.to_string/2`.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]:
```suggestion
Но что, если у вас есть огромный кусок AST, и вы хотите в нём разобраться? В таком случае вы можете использовать `Macro.to_string/2`.
```

File: ru/lessons/advanced/metaprogramming.md [hunk 7]
```diff
@@ -127,13 +144,69 @@ def test do
...
+Рассмотрим следующий пример:
+
+```elixir
+iex> Macro.to_string(quote(do: foo.bar(1, 2, 3)))
+"foo.bar(1, 2, 3)"
+```
+
+А когда вы хотите посмотреть на код, сгенерированный макросами, можете сочетать их с `Macro.expand/2` и `Macro.expand_once/2`, эти функции расширяют макросы в соответствующий им код.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]:
```suggestion
А когда вы хотите посмотреть на код, сгенерированный макросами, можете сочетать их с `Macro.expand/2` и `Macro.expand_once/2`. Эти функции расширяют макросы в соответствующий им код.
```

File: ru/lessons/advanced/metaprogramming.md [hunk 8]
```diff
@@ -127,13 +144,69 @@ def test do
...
+Первая функция может расширять макрос несколько раз, вторая - только один.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]:
```suggestion
Первая функция может расширять макрос несколько раз, а вторая — только один.
```

File: ru/lessons/advanced/metaprogramming.md [hunk 9]
```diff
@@ -127,13 +144,69 @@ def test do
...
+iex> quoted |> Macro.expand(__ENV__) |> Macro.to_string |> IO.puts
+case(!true) do
+  x when x in [false, nil] ->
+    nil
+  _ ->
+    "Hi"
+end
+```
+
+Помните, мы упоминали: `if` - это макрос в Elixir, а мы расширили его в соответствующее выражение `case`.
```
Comment (nscyclone) [inline ru/lessons/advanced/metaprogramming.md]:
```suggestion
Помните, ранее мы упоминали `if`? В Elixir это макрос, а мы расширили его в соответствующее выражение `case`.
```

**Comment by filalex77** [comment]
@nscyclone Thanks for the detailed review! I've applied the suggested changes.

**Pull request merged**